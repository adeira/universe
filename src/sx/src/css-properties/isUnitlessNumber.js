// @flow strict

/**
 * CSS properties which accept numbers but are not in units of "px".
 * https://github.com/facebook/react/blob/87b3e2d257e49b6d2c8e662830fc8f3c7d62f85f/packages/react-dom/src/shared/CSSProperty.js
 */
module.exports = {
  // SX specific start
  fontSize: true,
  // SX specific end

  animationIterationCount: true,
  borderImageOutset: true,
  borderImageSlice: true,
  borderImageWidth: true,
  boxFlex: true,
  boxFlexGroup: true,
  boxOrdinalGroup: true,
  columnCount: true,
  columns: true,
  flex: true,
  flexGrow: true,
  flexPositive: true,
  flexShrink: true,
  flexNegative: true,
  flexOrder: true,
  gridArea: true,
  gridRow: true,
  gridRowEnd: true,
  gridRowSpan: true,
  gridRowStart: true,
  gridColumn: true,
  gridColumnEnd: true,
  gridColumnSpan: true,
  gridColumnStart: true,
  fontWeight: true,
  lineClamp: true,
  lineHeight: true,
  opacity: true,
  order: true,
  orphans: true,
  tabSize: true,
  widows: true,
  zIndex: true,
  zoom: true,

  // SVG-related properties
  fillOpacity: true,
  floodOpacity: true,
  stopOpacity: true,
  strokeDasharray: true,
  strokeDashoffset: true,
  strokeMiterlimit: true,
  strokeOpacity: true,
  strokeWidth: true,
};
