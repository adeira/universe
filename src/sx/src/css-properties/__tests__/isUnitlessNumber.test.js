// @flow strict

import isUnitlessNumber from '../isUnitlessNumber';

it('returns all unitless properties', () => {
  expect(isUnitlessNumber).toMatchSnapshot();
});
