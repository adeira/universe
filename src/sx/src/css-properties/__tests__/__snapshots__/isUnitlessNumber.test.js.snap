// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all unitless properties 1`] = `
{
  "animationIterationCount": true,
  "borderImageOutset": true,
  "borderImageSlice": true,
  "borderImageWidth": true,
  "boxFlex": true,
  "boxFlexGroup": true,
  "boxOrdinalGroup": true,
  "columnCount": true,
  "columns": true,
  "fillOpacity": true,
  "flex": true,
  "flexGrow": true,
  "flexNegative": true,
  "flexOrder": true,
  "flexPositive": true,
  "flexShrink": true,
  "floodOpacity": true,
  "fontWeight": true,
  "gridArea": true,
  "gridColumn": true,
  "gridColumnEnd": true,
  "gridColumnSpan": true,
  "gridColumnStart": true,
  "gridRow": true,
  "gridRowEnd": true,
  "gridRowSpan": true,
  "gridRowStart": true,
  "lineClamp": true,
  "lineHeight": true,
  "opacity": true,
  "order": true,
  "orphans": true,
  "stopOpacity": true,
  "strokeDasharray": true,
  "strokeDashoffset": true,
  "strokeMiterlimit": true,
  "strokeOpacity": true,
  "strokeWidth": true,
  "tabSize": true,
  "webkitAnimationIterationCount": true,
  "webkitBorderImageOutset": true,
  "webkitBorderImageSlice": true,
  "webkitBorderImageWidth": true,
  "webkitBoxFlex": true,
  "webkitBoxFlexGroup": true,
  "webkitBoxOrdinalGroup": true,
  "webkitColumnCount": true,
  "webkitColumns": true,
  "webkitFillOpacity": true,
  "webkitFlex": true,
  "webkitFlexGrow": true,
  "webkitFlexNegative": true,
  "webkitFlexOrder": true,
  "webkitFlexPositive": true,
  "webkitFlexShrink": true,
  "webkitFloodOpacity": true,
  "webkitFontWeight": true,
  "webkitGridArea": true,
  "webkitGridColumn": true,
  "webkitGridColumnEnd": true,
  "webkitGridColumnSpan": true,
  "webkitGridColumnStart": true,
  "webkitGridRow": true,
  "webkitGridRowEnd": true,
  "webkitGridRowSpan": true,
  "webkitGridRowStart": true,
  "webkitLineClamp": true,
  "webkitLineHeight": true,
  "webkitOpacity": true,
  "webkitOrder": true,
  "webkitOrphans": true,
  "webkitStopOpacity": true,
  "webkitStrokeDasharray": true,
  "webkitStrokeDashoffset": true,
  "webkitStrokeMiterlimit": true,
  "webkitStrokeOpacity": true,
  "webkitStrokeWidth": true,
  "webkitTabSize": true,
  "webkitWidows": true,
  "webkitZIndex": true,
  "webkitZoom": true,
  "widows": true,
  "zIndex": true,
  "zoom": true,
}
`;
